src/v/datalake/tests/record_multiplexer_bench.cc

```diff
@@ -343,7 +340,7 @@ class record_multiplexer_bench_fixture
         }
 
         auto reg_res = co_await _record_gen.register_avro_schema(name, schema);
-        BOOST_REQUIRE(reg_res.has_value());
+        vassert(reg_res.has_value(), "must register avro schema for benchmark");
```

Comment:
All these vassert changes are outside of the benched region right?

Reply:
Yes, it's in the setup code

Reply:
Actually the counting consumer is in the benched region, I'll just throw an exception instead.